compiler/parser/python.lalrpop

```diff
@@ -357,6 +358,336 @@ CompoundStatement: ast::Stmt = {
     ClassDef,
 };
 
+MatchStatement: ast::Stmt = {
+    <location:@L> "match" <subject:NamedExpressionTest> ":" "\n" Indent <cases:MatchCase+> Dedent <end_location:@R> => {
+        ast::Stmt {
+            custom: (),
+            location,
+            end_location: Some(end_location),
+            node: ast::StmtKind::Match {
+                subject: Box::new(subject),
+                cases
+            }
+        }
+    }
+}
+
+MatchCase: ast::MatchCase = {
+    "case" <pattern:Pattern> <guard:(Guard)?> ":" <body:Suite> => {
+        ast::MatchCase {
+            pattern,
+            guard: guard.map(Box::new),
+            body
+        }
+    },
+}
+
+Guard: ast::Expr = {
+    "if" <guard:NamedExpressionTest> => {
+        guard
+    }
+}
+
+Pattern: ast::Pattern = {
+    <pattern:AsPattern> => pattern,
+    <pattern:OrPattern> => pattern,
+}
+
+AsPattern: ast::Pattern = {
+   <location:@L> <name:Identifier> <end_location:@R> => ast::Pattern {
```

Comment:
isn't `'as'` allowed here? I.e, probably two alternatives where either a single ident is allowed (as you've currently allowed) and a `<pattern:OrPattern> "as" <alias:Identifier>`.

Reply:
though I haven't tried to add this yet so lalrpop might not like it (though if it parses the current grammar it should parse that)

Reply:
Yes, oops.

Reply:
(It's just one case here, the no-`as` case is a case on `ClosedPattern`. Have it working locally.)